nixos/modules/services/networking/birdsitelive.nix

```
@@ -0,0 +1,442 @@
+                    };
+                  };
+                };
+              };
+
+              Instance = {
+                Name = mkOption {
+                  type = types.str;
+                  default = "BirdsiteLIVE";
+                  description = mdDoc "Instance name.";
+                };
+
+                Domain = mkOption {
+                  type = types.str;
+                  default = config.networking.fqdn;
+                  defaultText = literalExpression "config.networking.fqdn";
+                  description = mdDoc "Domain name of the instance.";
+                };
+
+                AdminEmail = mkOption {
+                  type = types.str;
+                  description = mdDoc "Instance administrator email address.";
+                };
+
+                ResolveMentionsInProfiles = mkOption {
+                  type = types.bool;
+                  default = true;
+                  description = mdDoc "Whether to enable mentions parsing in Twitter profile descriptions.";
+                };
+
+                PublishReplies = mkOption {
+                  type = types.bool;
+                  default = false;
+                  description = mdDoc "Whether to publish replies.";
+                };
+
+                UnlistedTwitterAccounts = mkOption {
+                  type = with types; listOf str;
+                  default = [ ];
+                  description = mdDoc "List of Twitter account for which to enable unlisted publication.";
+                  apply = concatOrNull;
+                };
+
+                SensitiveTwitterAccounts = mkOption {
+                  type = with types; listOf str;
+                  default = [ ];
+                  description = mdDoc "List of Twitter account to mark all media from as sensitive by default.";
+                  apply = concatOrNull;
+                };
+
+                FailingTwitterUserCleanUpThreshold = mkOption {
+                  type = types.ints.unsigned;
+                  default = 700;
+                  description = mdDoc "Maximum error count before auto-removal of a Twitter account.";
+                };
+
+                FailingFollowerCleanUpThreshold = mkOption {
+                  type = types.ints.unsigned;
+                  default = 30000;
+                  description = mdDoc "Maximum error count before auto-removal of a Fediverse account.";
+                };
+
+                UserCacheCapacity = mkOption {
+                  type = types.ints.unsigned;
+                  default = 10000;
+                  description = mdDoc "Caching limit for Twitter user retrieval.";
+                };
+              };
+
+              Db = {
+                Type = mkOption {
+                  type = types.str;
+                  default = "postgres";
+                  visible = false;
+                  description = mdDoc "Database type.";
+                };
+
+                Host = mkOption {
```

```suggestion
              db = {
                type = mkOption {
                  type = types.str;
                  default = "postgres";
                  visible = false;
                  description = mdDoc "Database type.";
                };

                host = mkOption {
```
Please lowercase options if not using freeform settings.

---

These should be freeform settings.

---

The enclosing `services.birdsitelive.settings` option uses `freeformType` even though the explicitly listed options within use more concrete types. I could make the latter freeform as well, but that would only permit users specifying an incorrect configuration.

I don’t know the conventions around freeform settings, but sticking to the configuration structure of the application and defining explicit options with defaults, documentation and examples for mandatory and common items felt like the most sensible way to me.

---

freeform settings usually only list the most important settings and leave the less common used ones out but indeed in freeform they should be uppercase.